File: BattleBotServerPython/MotorHelper.py
#!/usr/bin/python
import RPi.GPIO as GPIO
GPIO.setmode(GPIO.BCM)
import math
class MotorHelper:
    
    def __init__(self, PWM, PWMchannel, PF, PB):
        '''
        This function requires the following parameters:
        - pwm: The servohat object.
        - GPIO: The gpio object.
        - PWMchannel: What pin on the Adafruit Servohat is for the PWM?
        - PF: What pin triggers forwards?
        - PB: What pin triggers backwards?
        '''
        self.pwm = PWM
        self.pwmchannel = PWMchannel;
        self.pf = PF
        self.pb = PB
        GPIO.setup(PF, GPIO.OUT)
        GPIO.setup(PB, GPIO.OUT)
    def Forward(self, speed):
        print "Forwards: "+str(self.mapp(speed))
        GPIO.output(self.pf, GPIO.HIGH)
        GPIO.output(self.pb, GPIO.LOW)
        self.pwm.set_pwm(self.pwmchannel, 0, self.mapp(speed))
    def Backward(self, speed):
        speed = speed * -1;
        print "Backwards: "+str(self.mapp(speed))
        GPIO.output(self.pf, GPIO.LOW)
        GPIO.output(self.pb, GPIO.HIGH)
        self.pwm.set_pwm(self.pwmchannel, 0, self.mapp(speed))
    def Stop(self):
        print "STOP"
        GPIO.output(self.pf, GPIO.LOW)
        GPIO.output(self.pb, GPIO.LOW)
        self.pwm.set_pwm(self.pwmchannel, 0, 0)

    def mapp(self,inputt):
        return int(math.floor(inputt * 4096 / 100));

File: BattleBotServerESP8266/MotorHelper.cpp
#include "MotorHelper.h"


MotorHelperClass::MotorHelperClass(int pwmpin, int PF, int PB)
{
	_pwmpin = pwmpin;
	_PF = PF;
	_PB = PB;
	pinMode(_PF, OUTPUT);
	pinMode(_PB, OUTPUT);
	pinMode(_pwmpin, OUTPUT);
}

void MotorHelperClass::Forward(int speed)
{
	digitalWrite(_PB, LOW);
	digitalWrite(_PF, HIGH);
	analogWrite(_pwmpin, map(speed, 0, 100, 0, 1024));
}

void MotorHelperClass::Backward(int speed)
{
	speed = speed * -1;
	digitalWrite(_PF, LOW);
	digitalWrite(_PB, HIGH);
	analogWrite(_pwmpin, map(speed, 0, 100, 0, 1024));
}

void MotorHelperClass::Stop()
{
	digitalWrite(_PF, LOW);
	digitalWrite(_PB, LOW);
	analogWrite(_pwmpin, 0);
}